classes/settings_provider.php

```diff
@@ -70,7 +75,7 @@ public static function get_quiz_element_types() : array {
             'seb' => 'header',
             'seb_requiresafeexambrowser' => ['select', self::get_requiresafeexambrowser_options()],
             'seb_sebconfigtemplate' => ['select', self::get_template_options()],
-            'seb_sebconfigfile' => 'filepicker',
+            'filemanager_sebconfigfile' => ['filemanager', self::get_filemanager_options()],
```

Comment:
Is name important here for moodle API? If not I'd suggest to rename for consistency all other are like seb_***

Reply:
The name was changed here as to not conflict with the existing quiz_settings persistent property 'sebconfigfile'.

Once the hook to add modform elements has finished, a $mform->set_data(...) function is called and overwrites the form element 'seb_sebconfigfile'. Which then prevented displaying the saved config file from the module context.

Reply:
Also the mod_form hook for quiz access plugins did not allow for other moodle form hooks to be called, eg, data_preprocessing, definition_after_data etc.
```
mod_form.php:346
// Any other rule plugins.
quiz_access_manager::add_settings_form_fields($this, $mform);
```
With this, setting the data with a set_data() call was failing because the form was not even constructed completely at that time. 

Given that seb_configfile was being saved, and was part of the persistent class, when the saving the modform data, an eventual setdata call was made and it just restored the users draft area, or no items at all.

To get around this I made this exception. Though I'd love to know how we can keep it generic and not require this at all.